File: flutter-sdk/lib/src/data/services/yalo_message_auth/yalo_message_auth_service.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:yalo_chat_flutter_sdk/src/common/result.dart';
import 'package:yalo_chat_flutter_sdk/src/data/services/yalo_message_auth/token_entry.dart';

abstract class YaloMessageAuthService {
  // Auth calls the /auth endpoint and caches the token.
  // If the token is expired it renews it with the refresh token.
  Future<Result<TokenEntry>> auth();
}

File: android-sdk/sdk/src/main/kotlin/com/yalo/chat/sdk/data/services/auth/YaloMessageAuthService.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.data.services.auth

import com.yalo.chat.sdk.common.Result

// Port of flutter-sdk YaloMessageAuthService (data/services/auth).
// Calls POST /auth for anonymous token acquisition and POST /oauth/token for refresh.
// All results are wrapped in Result<T>; no exceptions are thrown to the caller.
interface YaloMessageAuthService {
    // Returns a valid access token.
    // On first call (no cache) — POSTs to /auth.
    // On subsequent calls — returns cached token if still valid, or POSTs to /oauth/token to refresh.
    suspend fun auth(): Result<String>
}
